### PR TITLE
Fix window dialog drag expanding to the opposite direction

### PR DESCRIPTION
**Pull request by 84634E1A607A**
**Brief Description of What This PR Does**

This PR:

https://user-images.githubusercontent.com/60734649/149648278-b3937eba-32d0-4e03-9904-b3295a375338.mp4

Master:

https://user-images.githubusercontent.com/60734649/149648282-de399a21-6e3f-48fd-96b3-2d679166fc3f.mp4

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/gui_common/CustomDialog.cs [hunk 1]
```diff
@@ -436,31 +436,32 @@ private void HandleActiveDrag()
         else
         {
             // Handle border dragging
+            var screenSize = (GetViewport()?.GetVisibleRect().Size).GetValueOrDefault();
```
Comment (hhyyrylainen) [inline src/gui_common/CustomDialog.cs:439]: I think we should print an error an exit this method if the viewport doesn't exist. I'm pretty sure I've done pretty hard errors in the past about missing critical objects. The viewport can only be missing if this isn't attached to the scene tree so this wouldn't be able to receive the _GuiInput event anyway in that case.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/CustomDialog.cs (hunk 1 above)
Comment (84634E1A607A) [reply src/gui_common/CustomDialog.cs:439]: I just copied this from the function below. So maybe we can just remove it.

Comment (hhyyrylainen) [reply src/gui_common/CustomDialog.cs:439]: Probably. Might also want to move this to a helper method as this is pretty dense logic that doesn't make much sense if it is edited in one place and not the other (even if that happens accidentally).

**Review by Kasterisk — APPROVED** [review]
Nothing mildly concerning other than what's mentioned by hh.

**Review by RacerBG — APPROVED** [review]
LGTM!

**Comment by 84634E1A607A** [comment]
I just realized I should have used `Mathf.Clamp` instead of nesting `Mathf.Max` and `Mathf.Min`. Will change that later.

**Review by hhyyrylainen — APPROVED** [review]
I did a quick test and at least the auto-evo predictions popup box worked fine. Code looks fine now and I assume that the people who I requested re-reviews would be fine as well with this still even though they didn't re-review.

**Pull request merged**